File: Daily Test/Java/Video Gaming Session 29.5.2025.java
import java.util.*;

public class Main {
    public static int toMinutes(String time) {
        String[] parts = time.split(":");
        return Integer.parseInt(parts[0]) * 60 + Integer.parseInt(parts[1]);
    }

    public static String toTime(int mins) {
        return String.format("%02d:%02d", mins / 60, mins % 60);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.next();
        String e = sc.next();
        int x = sc.nextInt();
        int y = sc.nextInt();

        int start = toMinutes(s);
        int end = toMinutes(e);
        boolean printed = false;

        while (start + x <= end) {
            System.out.println(toTime(start) + "-" + toTime(start + x));
            start += x + y;
            printed = true;
        }

        if (!printed) System.out.println("-1");
    }
}

File: Daily Test/C/Video Gaming Session 29.5.2025.c
#include <stdio.h>

int toMinutes(char *time) {
    int h, m;
    sscanf(time, "%d:%d", &h, &m);
    return h * 60 + m;
}

void printTime(int mins) {
    printf("%02d:%02d", mins / 60, mins % 60);
}

int main() {
    char s[6], e[6];
    int x, y;
    scanf("%s %s", s, e);
    scanf("%d %d", &x, &y);

    int start = toMinutes(s);
    int end = toMinutes(e), next;

    int printed = 0;
    while ((next = start + x) <= end) {
        printTime(start);
        printf("-");
        printTime(next);
        printf("\n");
        start = next + y;
        printed = 1;
    }

    if (!printed)
        printf("-1\n");
    return 0;
}